src/Http/WebUtilities/src/FileBufferingReadStream.cs

```diff
@@ -223,13 +234,19 @@ public override int Read(byte[] buffer, int offset, int count)
                 {
                     oldBuffer.Position = 0;
                     var rentedBuffer = _bytePool.Rent(Math.Min((int)oldBuffer.Length, _maxRentedBufferSize));
-                    var copyRead = oldBuffer.Read(rentedBuffer, 0, rentedBuffer.Length);
```

Comment:
`_buffer.Write` may throw. Putting the renting inside of a try-finally block.

Reply:
Add a test?